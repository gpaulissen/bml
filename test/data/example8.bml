// https://github.com/gpaulissen/bml/issues/4
// strange characters for laTeX: & % $ # _ { } ~ ^ \

1N-2C PUP to 2!d
  2D Forced
    P Weak with !d
    2H TRF
      2S
        2N 5+c !s, G/T
        3CD !s & minor, G/T (%$#_{}~^\)
