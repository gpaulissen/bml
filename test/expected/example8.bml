#VUL 00

#SEAT 0

1N
  2C = PUP to 2!d
    2D = *Forced*
      P = Weak with !d
      2H = TRF
        2S
          2N = 5+c !s, G/T
          3C = !s & minor, G/T (%$#_{}~^\)
          3D = !s & minor, G/T (%$#_{}~^\)
1C
  (1D)
    2C = support
    2D = cuebid
    2H = weak jump
    2S = weak jump
  (1H)
    2C = support
    2D = forcing
    2H = cuebid
    2S = weak jump
  (1S)
    2C = support
    2D = forcing
    2H = forcing
    2S = cuebid
