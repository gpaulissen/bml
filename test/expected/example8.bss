*00{}=NYYYYYY
001N=NYYYYYY0
001NP2C=NYYYYYY008PUP to 2!d
001NP2CP2D=NYYYYYY008*Forced*
001NP2CP2DPP=NYYYYYY0Weak with !d
001NP2CP2DP2H=NYYYYYY008TRF
001NP2CP2DP2HP2S=NYYYYYY008
001NP2CP2DP2HP2SP2N=NYYYYYY05+c !s, G/T
001NP2CP2DP2HP2SP3C=NYYYYYY008!s & minor, G/T (%$#_{}~^\)
001NP2CP2DP2HP2SP3D=NYYYYYY008!s & minor, G/T (%$#_{}~^\)
001C=NYYYYYY008
001C1D=NYYYYYY008
001C1D2C=NYYYYYY008support
001C1D2D=NYYYYYY008cuebid
001C1D2H=NYYYYYY008weak jump
001C1D2S=NYYYYYY008weak jump
001C1H=NYYYYYY008
001C1H2C=NYYYYYY008support
001C1H2D=NYYYYYY008forcing
001C1H2H=NYYYYYY008cuebid
001C1H2S=NYYYYYY008weak jump
001C1S=NYYYYYY008
001C1S2C=NYYYYYY008support
001C1S2D=NYYYYYY008forcing
001C1S2H=NYYYYYY008forcing
001C1S2S=NYYYYYY008cuebid
