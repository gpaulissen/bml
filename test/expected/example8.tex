\documentclass[a4paper]{article}
\usepackage[margin=1in]{geometry}
\usepackage[T1]{fontenc}
\usepackage[utf8]{inputenc}
\usepackage{newcent}
\usepackage{helvet}
\usepackage{graphicx}

\usepackage[pdftex, pdfborder={0 0 0}]{hyperref}
\frenchspacing

\usepackage{txfonts} % For \varheartsuit and \vardiamondsuit
\usepackage[usenames,dvipsnames]{color} % dvipsnames necessary to made PDFLaTeX work.
\usepackage{enumerate}
\usepackage{listliketab}
\usepackage{latexsym} % \Box
\usepackage{pbox} % \Box
\usepackage{parskip} % line between paragraphs

% suits

%%% Colors

\newcommand{\BC}{\textcolor{OliveGreen}{$\clubsuit$}}
\newcommand{\BD}{\textcolor{RedOrange}{$\vardiamondsuit$}}
\newcommand{\BH}{\textcolor{Red}{$\varheartsuit${}}}
\newcommand{\BS}{\textcolor{Blue}{$\spadesuit${}}}

%suits for pdf-friendly titles
\newcommand{\pdfc}{\texorpdfstring{\BC{}}{C}}
\newcommand{\pdfd}{\texorpdfstring{\BD{}}{D}}
\newcommand{\pdfh}{\texorpdfstring{\BH{}}{H}}
\newcommand{\pdfs}{\texorpdfstring{\BS{}}{S}}

\newenvironment{bidtable}
{\begin{tabbing}

xxxxxx\=xxxxxx\=xxxxxx\=xxxxxx\=xxxxxx\=xxxxxx\=xxxxxx\=xxxxxx\=xxxxxx\=xxxxxx\=\kill}
{\end{tabbing} }%

\newenvironment{bidding}%
{\begin{tabbing}
xxxxxx\=xxxxxx\=xxxxxx\=xxxxxx \kill
}{\end{tabbing} }%end bidding


% writing hands
\newcommand{\cards}[1]{\textsf{#1}}
\newcommand{\spades}[1]{\BS\cards{#1}}
\newcommand{\hearts}[1]{\BH\cards{#1}}
\newcommand{\diamonds}[1]{\BD\cards{#1}}
\newcommand{\clubs}[1]{\BC\cards{#1}}
\newcommand{\void}{--}
\newcommand{\hand}[4]{\spades{#1}\ \hearts{#2}\ \diamonds{#3} \clubs{#4}}
\newcommand{\vhand}[4]{\spades{#1}\\\hearts{#2}\\\diamonds{#3}\\\clubs{#4}}

% The \Box should always appear the same distance from the left margin
\newcommand\onesuit[4]%
{%
  \begin{center}%
    \begin{tabular}{>{\hfill}p{3cm}cp{3cm}}
                & \cards{#2} \\%
      \cards{#1}& $\Box$    & \cards{#3} \\%
                & \cards{#4} %
    \end{tabular}
  \end{center}%
}

% A special command if the south hand is not shown to avoid whitespace
\newcommand\onesuitenw[3]%
{%
  \begin{center}%
    \begin{tabular}{>{\hfill}p{3cm}cp{3cm}}%
                & \cards{#2} \\%
      \cards{#1}& $\Box$    & \cards{#3}%
    \end{tabular}%
  \end{center}%
}

\newcommand\dealdiagram[5]%
{%
  \begin{center}%
    \begin{tabular}{>{\hfill}p{3cm}cp{3cm}}
      \pbox{20cm}{\small #5}& \pbox{20cm}{#2} \\%
      \pbox{20cm}{#1}& $\Box$    & \pbox{20cm}{#3} \\%
                & \pbox{20cm}{#4} %
    \end{tabular}
  \end{center}%
}

\newcommand\dealdiagramenw[4]%
{%
  \begin{center}%
    \begin{tabular}{>{\hfill}p{3cm}cp{3cm}}
      \pbox{20cm}{\small #4}& \pbox{20cm}{#2} \\%
      \pbox{20cm}{#1}& $\Box$    & \pbox{20cm}{#3} \\%
    \end{tabular}
  \end{center}%
}

\newcommand\dealdiagramew[2]%
{%
  \begin{center}%
    \begin{tabular}{>{\hfill}p{3cm}cp{3cm}}
      \pbox{20cm}{#1}& $\Box$    & \pbox{20cm}{#2} \\%
    \end{tabular}
  \end{center}%
}

\title{No title defined}
\author{No author defined}
\begin{document}
\maketitle
\tableofcontents

\begin{bidtable}
1NT-2\BC \> PUP to 2\BD \+\\
2\BD \> *Forced*\+\\
Pass \> Weak with \BD \\
2\BH \> TRF\+\\
2\BS\+\\
2NT \> 5+c \BS , G/T\\
3\BC\BD \> \BS\ \& minor, G/T (\%\$\#\_\{\}\textasciitilde\textasciicircum\textbackslash)\-\-\-\-
\end{bidtable}
\bigbreak
\begin{bidtable}
1\BC-(1Y)\+\\
2\BC \> support\\
2X \> forcing\\
2Y \> cuebid\\
2Z \> weak jump\-
\end{bidtable}
\bigbreak
\end{document}
